runtime/common/src/lib.rs

```diff
@@ -103,9 +104,21 @@ where
 {
 	let input_prices: PriceCollectionInput<T> =
 		if let Ok(prices) = pallet_loans::Pallet::<T>::registered_prices(pool_id) {
-			PriceCollectionInput::Custom(prices.try_into().map_err(|_| {
-				DispatchError::Other("Map expected to fit as it is coming from loans itself.")
-			})?)
+			PriceCollectionInput::Custom(
```

Comment:
Not sure if I'm missing something but this entire block is not just the same as: `PriceCollectionInput::FromRegistry`?

Reply:
No, unfortuntealy we need this in order to decide ourselves to only use the registry sources if it is not failing. If it is failing, we are going with empty prices. If we forwared `FromRegistry` then the whole call would fail.

Reply:
But just for the case where we're sure it doesn't fail. We can add `FromRegistry`

Reply:
```rust
if let Ok(prices) = pallet_loans::Pallet::<T>::registered_prices(pool_id) {
    PriceCollectionInput::FromRegistry,
} else {
    PriceCollectionInput::Empty
};
```

Reply:
Nice!